main.js

```diff
@@ -179,14 +179,32 @@ function reviveClientApp(initialAppConfigFunctions) {
   });
 }
 
+function isDocumentInteractive() {
+  return ['interactive', 'complete'].includes(document.readyState);
+}
+
 function onLoad() {
   vendorLinker.bindToNamespace(ns);
 
   if (!_isClient()) {
     return Promise.reject(null);
   }
 
-  return Promise.resolve();
+  if (isDocumentInteractive()) {
+    return Promise.resolve();
+  }
+
+  return new Promise(resolve => {
+    document.addEventListener(
+      'readystatechange',
```

Comment:
The event DomContentLoaded has got better support in browser https://caniuse.com/#feat=domcontentloaded